File: py/198.py
# Author: Andrew Hamara

# Solution to leetcode problem 198. House Robber

class Solution:
    def rob(self, nums:List[int]) -> int:
        twoAgo = 0
        lastRobbed = 0
        currentRob = -10000

        for n in nums:
            currentRob = max(twoAgo + n, lastRobbed)
            twoAgo = lastRobbed
            lastRobbed = currentRob
        return currentRob

File: go/198.go
func rob(nums []int) int {
    var lastRobbed int = 0
    var twoAgo int = 0
    var curRob int = -1000000

    for _, n := range nums {
        if n + twoAgo > lastRobbed {
            curRob = n + twoAgo
        }else {
            curRob = lastRobbed
        }
        twoAgo = lastRobbed
        lastRobbed = curRob
    }
    return curRob
}
